crates/wasmtime/src/runtime/vm/component/libcalls.rs

```diff
@@ -569,3 +569,7 @@ unsafe fn resource_enter_call(vmctx: *mut VMComponentContext) -> Result<()> {
 unsafe fn resource_exit_call(vmctx: *mut VMComponentContext) -> Result<()> {
     ComponentInstance::from_vmctx(vmctx, |instance| instance.resource_exit_call())
 }
+
+unsafe fn trap(_vmctx: *mut VMComponentContext, code: u32) -> Result<()> {
+    Err(wasmtime_environ::Trap::from_u8(code.try_into().unwrap()).unwrap()).err2anyhow()
```

Comment:
Can this take a `code: u8` instead of a `code: u32` if it is just going to `code.try_into().unwrap()` it? Is there a reason for the weird mismatch of bit widths?

Reply:
Nah just more effort to plumb u8 to more locations, but I've now done that.